src/structures/cluster.cpp

```diff
@@ -80,4 +80,22 @@ int32_t Cluster::get_average_inserted_sequence_size() const
     }
     int32_t average_size = std::round(static_cast<double>(sum_sizes) / members.size());
     return average_size;
+}
+
+std::vector<Junction> Cluster::get_members() const
+{
+    return members;
+}
+
+bool operator<(Cluster const & lhs, Cluster const & rhs)
+{
+    return lhs.get_average_mate1() < rhs.get_average_mate1()
+            ? true
+            : rhs.get_average_mate1() < lhs.get_average_mate1()
+                ? false
+                : lhs.get_average_mate2() < rhs.get_average_mate2()
+                    ? true
+                    : rhs.get_average_mate2() < lhs.get_average_mate2()
+                        ? false
+                        : lhs.get_average_inserted_sequence_size() < rhs.get_average_inserted_sequence_size();
```

Comment:
Interesting... is this really the best we can do with c++? I just searched for it, but I couldn't find any better way...

Reply:
There is actually another but still similar way which is used in `breakend.cpp`. It first checks for inequality first and then checks which is smaller. Maybe that way is more understandable?!

```
    return lhs.seq_name != rhs.seq_name
            ? lhs.seq_name < rhs.seq_name
            : lhs.orientation != rhs.orientation
                ? lhs.orientation < rhs.orientation
                : lhs.position < rhs.position;
```

Reply:
I like your suggestion. It is much clearer.

I think if you can use fields instead of function you can also do the following:
```
return std::tie(lhs.seq_name, lhs.orientation, lhs.position) < std::tie(rhs.seq_name, rhs.orientation, rhs.position);
```
But if you use function it might copy and be slower :-(

Reply:
I'm not sure I understand what you mean with using "fields instead of function"  (can you maybe explain a bit more?) but the expression you suggest looks much cleaner and shorter. According to the C++ reference, `std::tie` creates a tuple of references so it should be only a small overhead compared to using the const references in its present form.

Reply:
> I'm not sure I understand what you mean with using "fields instead of function" (can you maybe explain a bit more?) but the expression you suggest looks much cleaner and shorter. According to the C++ reference, `std::tie` creates a tuple of references so it should be only a small overhead compared to using the const references in its present form.

Yeah, I wasn't very precise.
If you have for example `return std::tie(lhs.func1(), lhs.func2()) < std::tie(rhs.func1(), rhs.func2())`  All 4 functions have to be evaluated before the `<`-operator is being executed. If  `func1` and `func2` are very slow functions this might not be what you want and you want to use your original solution, which only executes `func1` and only if they are the same executes `func2`.
If you use variables (I called them accidentally fields) instead of functions, you don't have to worry about slow performing functions. In general this is all microoptimization. I got the feeling I am derailing your PR :disappointed: .
Any of your suggestions are fine. :smiley:

Reply:
Thanks for the explanation. Yes, that makes a lot of sense and I think it's always good to keep these smaller optimizations in mind 👍  So no worries, I'm very thankful for this kind of discussion 😃 

I now reimplemented the comparison operator for the `breakend` class because it accesses only member variables. However, I left the operators of the `junction` and `cluster` classes because these use functions which create additional `breakend` copies.